### PR TITLE
docs(storage): add GCS+gRPC release notes

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11760)

### CODE REVIEW

File: google/cloud/storage/doc/storage-grpc.dox [hunk 1]
```diff
@@ -5,8 +5,40 @@
 The GCS C++ client library includes an optional plugin to access GCS via gRPC.
 When using GCS from Google Compute Engine (GCE) this plugin can enable higher
 total throughput across large workloads that run on hundreds or thousands of
-VMs.  Note that this feature is not GA, consult with with account manager to
-be included in the early access.
+VMs.  Note that this feature is not GA, consult with with you account manager to
```
Comment (dbolduc) [inline google/cloud/storage/doc/storage-grpc.dox:8]: s/GA, consult/GA. consult/
s/with with you/with your/

File: google/cloud/storage/doc/storage-grpc.dox [hunk 2]
```diff
@@ -5,8 +5,40 @@
...
+be included in the early access program.
+
+## Release Notes
+
+There are some features that the gRPC plugin does not implement or implements
+differently from the REST-based implementation.
+
+1. For best performance consider using Protobuf >= 23.2 and gRPC >= 1.55.0.
+   Older versions of Protobuf requires enable [ctype=CORD workarounds] at
```
Comment (dbolduc) [inline google/cloud/storage/doc/storage-grpc.dox:17]: s/enable/enabling/

File: google/cloud/storage/doc/storage-grpc.dox [hunk 3]
```diff
@@ -5,8 +5,40 @@
...
+   compile-time.
+1. To override the default endpoint you should use
+   [`EndpointOption`](@ref google::cloud::EndpointOption) instead of
+   [`RestEndpointOption`](@ ref google::cloud::storage::RestEndpointOption).
+1. The default endpoint (`storage.googleapis.com`) works from any hosting
+   environment (onprem, GKE, GCE, other clour providers, etc.). For best
```
Comment (dbolduc) [inline google/cloud/storage/doc/storage-grpc.dox:23]: s/onprem/on-prem/ ? (maybe)
s/clour/cloud/

File: google/cloud/storage/doc/storage-grpc.dox [hunk 4]
```diff
@@ -5,8 +5,40 @@
...
+   performance on GCE or GKE consider using
+   `google-c2p:///storage.googleapis.com`.
+1. The [`Fields`](@ref google::cloud::storage::Fields) request parameter is
+   passed verbatim to the backend. Some fields have different names in gRPC, and
+   the backend does not translate them. This is a rarely used feature, and we do
+   not anticipate the behavior change is likely to cause problems. If it does
+   cause any issues please [file a bug].
+1. All the `storage::Client::*AccessControl()` functions are emulated. For
+   functions that change state the emulation uses a [OCC loop]. While this
```
Comment (dbolduc) [inline google/cloud/storage/doc/storage-grpc.dox:32]: "that change state, the emulation uses an \[OCC loop\]."

**Review by dbolduc — APPROVED** [review]

File: google/cloud/storage/doc/storage-grpc.dox [hunk 5]
```diff
@@ -5,8 +5,40 @@
 The GCS C++ client library includes an optional plugin to access GCS via gRPC.
 When using GCS from Google Compute Engine (GCE) this plugin can enable higher
 total throughput across large workloads that run on hundreds or thousands of
-VMs.  Note that this feature is not GA, consult with with account manager to
-be included in the early access.
+VMs.  Note that this feature is not GA. Consult with with your account manager
```
Comment (dbolduc) [inline google/cloud/storage/doc/storage-grpc.dox:8]: s/with with/with/

File: google/cloud/storage/doc/storage-grpc.dox [hunk 6]
```diff
@@ -29,7 +29,7 @@ differently from the REST-based implementation.
    not anticipate the behavior change is likely to cause problems. If it does
    cause any issues please [file a bug].
 1. All the `storage::Client::*AccessControl()` functions are emulated. For
-   functions that change state the emulation uses a [OCC loop]. While this
+   functions that change state, the emulation uses a [OCC loop]. While this
```
Comment (dbolduc) [inline google/cloud/storage/doc/storage-grpc.dox:32]: s/uses a/uses an/

File: google/cloud/storage/doc/storage-grpc.dox (hunk 5 above)
Comment (coryan) [reply google/cloud/storage/doc/storage-grpc.dox]: Argh, sorry I missed that

File: google/cloud/storage/doc/storage-grpc.dox (hunk 6 above)
Comment (coryan) [reply google/cloud/storage/doc/storage-grpc.dox]: Fixed

File: google/cloud/storage/doc/storage-grpc.dox (hunk 5 above)
Comment (coryan) [reply google/cloud/storage/doc/storage-grpc.dox]: Argh, sorry I missed that

File: google/cloud/storage/doc/storage-grpc.dox (hunk 6 above)
Comment (coryan) [reply google/cloud/storage/doc/storage-grpc.dox]: Fixed

**Pull request merged**